### PR TITLE
[Train]  Handle Arrow-backed pandas dtypes in LightGBM examples

### PR DESCRIPTION
**Pull request by pseudo-rnd-thoughts**
## Description

https://github.com/ray-project/ray/pull/63017 updated Ray Data's Arrow-to-pandas conversion to preserve Arrow-backed pandas dtypes, such as `int64[pyarrow]`, so dtypes can roundtrip more faithfully.

This exposed an incompatibility with LightGBM's pandas input path. Ray Train's LightGBM examples and legacy trainer code convert Ray Data shards to pandas before constructing `lightgbm.Dataset`. With Arrow-backed Ray Data inputs, those pandas DataFrames can now contain Arrow-backed dtypes, and LightGBM rejects them during pandas dtype validation even when the logical column type is numeric.

This PR updates the LightGBM paths to normalize pandas DataFrames to NumPy-nullable pandas dtypes before passing them to LightGBM. It also updates documentation and examples to show the same conversion for user-authored LightGBM training loops.

## Changes

- Normalize pandas DataFrames in the legacy `LightGBMTrainer` path before constructing `lightgbm.Dataset`.
- Update LightGBM docs and docstring examples to call `convert_dtypes(dtype_backend="numpy_nullable")` after `to_pandas()`.
- Add a regression test covering Arrow-backed Ray Data input from `ray.data.from_items(...)`.
- Keep the restore test focused on trainer restore behavior by using a pandas-backed test dataset.
- Update the LightGBM release benchmark to avoid passing Arrow-backed pandas dtypes to LightGBM.

### CODE REVIEW

File: python/ray/train/lightgbm/_lightgbm_utils.py [hunk 1]
```diff
@@ -12,6 +12,57 @@
...
+def normalize_pandas_for_lightgbm(df: "pd.DataFrame") -> "pd.DataFrame":
+    """Map Arrow-backed pandas dtypes to NumPy-nullable equivalents.
+
+    LightGBM's pandas input validation rejects Arrow-backed dtypes like
+    ``int64[pyarrow]``. Since Ray Data 2.56, ``Dataset.to_pandas()`` preserves
+    Arrow-backed dtypes when the source was Arrow, so callers passing the
+    resulting frame to ``lightgbm.Dataset`` must normalize first.
+
+    This helper is a faster alternative to
+    ``df.convert_dtypes(dtype_backend="numpy_nullable")``:
```
Comment (goutamvenkat-anyscale) [inline python/ray/train/lightgbm/_lightgbm_utils.py:29]: do we have benchmarks on this function comparison?

Comment (pseudo-rnd-thoughts) [reply python/ray/train/lightgbm/_lightgbm_utils.py:29]: Using lightgbm_train_batch_inference_benchmark_100G's release tests then I've tested both option which shows almost no difference so I would suspect that we don't have a good test to check this currently
```txt
normalize_pandas_for_lightgbm
real    4m35.544s
user    0m22.508s
sys     0m4.320s

df.convert_dtypes(dtype_backend="numpy_nullable")
real    4m35.533s
user    0m22.264s
sys     0m4.343s
```

**Review by goutamvenkat-anyscale — APPROVED** [review]

File: python/ray/train/lightgbm/_lightgbm_utils.py (hunk 1 above)
Comment (TimothySeah) [reply python/ray/train/lightgbm/_lightgbm_utils.py:29]: If that's the case should we either:
1) Just tell users to use `df.convert_dtypes(dtype_backend="numpy_nullable")` instead of adding a new API? That seems more stable than this API e.g. what if lightgbm starts accepting `timestamp` in the future?
2) If we still want this API, should we still call out that it's faster than `df.convert_dtypes(dtype_backend="numpy_nullable")` when the benchmark shows a negligible difference?

File: python/ray/train/lightgbm/_lightgbm_utils.py [hunk 2]
```diff
@@ -12,6 +12,57 @@
...
+    """
+    import pandas as pd
+    import pyarrow as pa
+
+    dtype_mapping = {}
+    for column, dtype in df.dtypes.items():
+        if not isinstance(dtype, pd.ArrowDtype):
+            continue
+        arrow_dtype = dtype.pyarrow_dtype
+        if pa.types.is_signed_integer(arrow_dtype):
```
Comment (TimothySeah) [inline python/ray/train/lightgbm/_lightgbm_utils.py:54]: nit: does https://arrow.apache.org/docs/python/generated/pyarrow.types.is_integer.html work or do we need to explicitly check both is_signed and is_unsigned?

**Review by TimothySeah — APPROVED** [review]
Approving to unblock for now - main question is why not just use `df.convert_dtypes(dtype_backend="numpy_nullable")`?

File: python/ray/train/lightgbm/_lightgbm_utils.py [hunk 3]
```diff
@@ -12,6 +12,57 @@
 from ray.tune.utils import flatten_dict
 from ray.util.annotations import PublicAPI
 
+if TYPE_CHECKING:
+    import pandas as pd
+
+
+@PublicAPI(stability="beta")
```
Comment (matthewdeng) [inline python/ray/train/lightgbm/_lightgbm_utils.py]:
```suggestion
@PublicAPI(stability="alpha")
```
